disttask/framework/storage/task_table.go

```diff
@@ -640,32 +640,39 @@ func (stm *TaskManager) UpdateGlobalTaskAndAddSubTasks(gTask *proto.Task, subtas
 			return err
 		}
 		if se.GetSessionVars().StmtCtx.AffectedRows() == 0 {
-			retryable = false
-			return errors.New("invalid task state transform, state already changed")
+			rs, err := ExecSQL(stm.ctx, se, "select id from  mysql.tidb_global_task where id = %? and state = %?", gTask.ID, prevState)
```

Comment:
Maybe gtask has no update(currently add index/import into will always change gtask), we need to check if the task have changed its state by admin cancel.

Reply:
why query it again when `StmtCtx.AffectedRows()` is 0?

Reply:
> why query it again when `StmtCtx.AffectedRows()` is 0?

If some onNextStage didn't update gtask, the affectedrows() return 0. But if the state didn't change by other admin command, it's safe to insert subtasks.
This case will not occurred for current add index and import into.